src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -7,22 +7,19 @@
 import java.util.List;
 
 public class MachineServiceImpl implements MachineService<Machine> {
-
     @Override
     public List<? extends Machine> getAll(Class type) {
+        List<? extends Machine> output = List.of();
         if (type.equals(Truck.class)) {
-            TruckProducer truckProducer = new TruckProducer();
-            return truckProducer.get();
+            output = new TruckProducer().get();
         }
         if (type.equals(Excavator.class)) {
-            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
-            return excavatorProducer.get();
+            output = new ExcavatorProducer().get();
         }
         if (type.equals(Bulldozer.class)) {
-            BulldozerProducer bulldozerProducer = new BulldozerProducer();
-            return bulldozerProducer.get();
+            output = new BulldozerProducer().get();
         }
-        return List.of();
+        return output;
```

Comment:
Your solution is fine, but let's use polymorphism and call the `get` method only once

Reply:
Sorry, this is too hard for me. idk how to do that